src/Credentials/GCECredentials.php

```diff
@@ -105,8 +110,11 @@ public static function onGce(ClientInterface $client = null)
       // could lead to false negatives in the event that we are on GCE, but
       // the metadata resolution was particularly slow. The latter case is
       // "unlikely".
-      $resp = $client->get($checkUri, ['timeout' => 0.3]);
-      return $resp->getHeader(self::FLAVOR_HEADER) == 'Google';
+      $resp = $httpHandler(
+        new Request('GET', $checkUri),
+        ['timeout' => 0.3]
```

Comment:
We can just throw it in the `README` in a section **Creating Custom Http Handlers** and then just mention the options used. At this time, I believe this is the only instance of the library using the options anyway.